system-tests/api-changing-settings-should-dump/src/test/java/com/teamscale/tia/client/ApiChangingSettingsShouldDumpSystemTest.java

```diff
@@ -0,0 +1,33 @@
+package com.teamscale.tia.client;
+
+import com.teamscale.test.commons.SystemTestUtils;
+import com.teamscale.test.commons.TeamscaleMockServer;
+import org.junit.jupiter.api.Test;
+import systemundertest.SystemUnderTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Runs the system under test and then changes the partition of the agent. This should cause a dump to our
+ * {@link TeamscaleMockServer}.
+ */
+public class ApiChangingSettingsShouldDumpSystemTest {
+
+	@Test
+	public void systemTest() throws Exception {
+		System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StdErrLog");
+		System.setProperty("org.eclipse.jetty.LEVEL", "OFF");
+
+		TeamscaleMockServer teamscaleMockServer = new TeamscaleMockServer(
+				SystemTestUtils.TEAMSCALE_PORT).acceptingReportUploads();
+
+		new SystemUnderTest().foo();
+		SystemTestUtils.changePartition(SystemTestUtils.AGENT_PORT, "some_other_value");
+
+		assertThat(teamscaleMockServer.uploadedReports).hasSize(1);
+		String report = teamscaleMockServer.uploadedReports.get(0).getReportString();
+		// ensure that only the foo() method was covered, not the bar() method
```

Comment:
Why would `bar` appear here? AFAICS it is never executed.

Reply:
it appears with `mi="2"`, i.e. all probes are missed in that line. So this, as the comment states, checks that the bar method was *not* covered. I tried to refactor the code to make this clearer. Let me know if it worked

Reply:
That's clear to me. I was just confused as to why we even check for that condition i.e. no matter if the dumping would be broken or not we would never expect bar to be covered so why do we check for that.